tools/osbuild-depsolve-pacman

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name
+
+"""
+A JSON-based interface for depsolving using Pacman.
+
+Reads a request through stdin and prints the result to stdout.
+In case of error, a structured error is printed to stdout as well.
+"""
+import json
+import os
+import os.path
+import sys
+
+from osbuild.solver import DepsolveError, GPGKeyReadError, InvalidRequestError, MarkingError, RepoError
+from osbuild.solver.pacman import Pacman as Solver
+
+
+def get_string_option(option):
+    # option.get_value() causes an error if it's unset for string values, so check if it's empty first
+    if option.empty():
+        return None
+    return option.get_value()
+
+
+def setup_cachedir(request):
+    arch = request["arch"]
+    # If dnf-json is run as a service, we don't want users to be able to set the cache
```

Comment:
dnf-json? :)

Reply:
Yea so, I'm just not sure about this and a lot of other things going on in this file for now. I'll revisit it.

Reply:
The depsolve request for the dnf depsolver (but also for the pacman depsolver) contains a cache key that configures the path for the cache.  In the service (in image builder) we set an `OVERWRITE_CACHE_DIR` env var to set a global cache and ignore the one in the request.  This is done to prevent users from setting the cache directory to something arbitrary or potentially destructive.

I think in the current state, this isn't possible (dnf-json / osbuild-depsolve-dnf isn't run as a service which is open to users), but at some point it did or was planned to be.